webhooks-extension/src/components/WebhookDisplayTable/WebhookDisplayTable.js

```diff
@@ -80,6 +135,94 @@ export class WebhookDisplayTable extends Component {
     }
   }
 
+  componentDidMount() {
+    this.getWebhooks();
+  }
+
+  async getWebhooks() {
+    const webhookData = await getWebhooks();
+    this.setState({
+      isLoaded: true,
+      webhooks: webhookData
+    });
+  }
+
+	showDeleteDialogHandlerInvisible = () => {
+		this.setState({
+			showDeleteDialog: false
+		});
+  }
+
+  showDeleteDialogHandlerVisible = rowsInput => {
+		this.setState({
+			showDeleteDialog: true,
+      userSelectedRows: rowsInput,
+    });
+	}
+
+  handleDeleteWebhook = () => {
+    let deleteRuns = false
+
+    if (this.state.checked) {
+        deleteRuns = true
+    };
+
+    let rowsToUse = getSelectedRows(this.state.userSelectedRows)
+
+    let deletePromises = []
+
+    deletePromises = rowsToUse.map(function(rowIDObject, indexInTheMap) {
+      let id = rowIDObject.id
+      let theName = id.substring(0, id.lastIndexOf('|'))
+      let namespace = id.substring(id.lastIndexOf('|') + 1, id.length)
+      let response = deleteWebhooks(theName, namespace, deleteRuns)
+      let theTimeoutPeriod = 500
```

Comment:
would the timeout need to scale depending on the number of webhooks selected ?

Reply:
It's for each webhook (or at least that's the intention!), I'll admit I haven't tried with lots of deletes. I'm going to add a comment here to mention this may be something we need to look into as we focus on scale and performance